cotainr/util.py

```diff
@@ -133,7 +133,14 @@ def stream_subprocess(*, args, log_dispatcher=None, **kwargs):
         stderr="".join(captured_stderr),
     )
 
-    completed_process.check_returncode()
+    try:
+        completed_process.check_returncode()
+    except subprocess.CalledProcessError as err:
+        print(f"A subprocess command {err.cmd} " +
```

Comment:
Here we should tap into the `cotainr.tracing` machinery (see the [cotainr dev docs tracing page](https://cotainr.readthedocs.io/en/stable/development/tracing_logging.html) for an introduction) for properly formatting the error message. Only if no `log_dispatcher` is passed to `stream_subprocess` should we directly `print`.

Ideally, I would like to use the `logging.expection` machinery here (see https://stackoverflow.com/a/5191885 for an example). I guess the cleanest solution involves implementing a `log_expection(...)` method in `cotainr.tracing.LogDispatcher` that is a thin wrapper around `LogDispatcher.logger_stderr.expection(...)`. This may potentially also involve modifying the formatting  (see https://docs.python.org/3/library/logging.html#logging.Formatter.format) of the error messages - but only if strictly necessary to provide an easily readable error messages from the subprocess.

Reply:
The `logging.exception` does not throw the original error message from the failed subprocess, which we want to display, but only the exception caught in the `except` statement, which is just the `CalledProcessError`. See that it can only be called inside an `except` statement.

Additionally, It seems as though the error message is already passed to the `log_dispatcher`. At least when I write a test of the caplog, the error message is there, but the same test with `log_dispatcher = None` does not seem to properly print to capsys. I don't know if the is just pytest not properly capturing the stdout and stderr from the `executor_submit()`?